src/librustc_codegen_ssa/back/write.rs

```diff
@@ -1244,10 +1244,10 @@ fn start_executing_work<B: ExtraBackendMethods>(
         while !codegen_done
             || running > 0
             || (!codegen_aborted
-                && (work_items.len() > 0
-                    || needs_fat_lto.len() > 0
-                    || needs_thin_lto.len() > 0
-                    || lto_import_only_modules.len() > 0
+                && (!work_items.is_empty()
+                    || !needs_fat_lto.is_empty()
+                    || !needs_thin_lto.is_empty()
+                    || !lto_import_only_modules.is_empty()
                     || main_thread_worker_state != MainThreadWorkerState::Idle))
```

Comment:
I think this will read better if we move the nots out, i.e., instead of (!A || !B || !C) do !(A && B && C)

Reply:
to be clear, I feel like the move to !is_empty here is otherwise hurting readability